packages/native-stack/src/types.tsx

```diff
@@ -682,6 +708,196 @@ export type NativeStackNavigationOptions = {
+    style?: {
+      fontFamily?: string;
+      fontSize?: number;
+      fontWeight?: string;
+    };
+  };
+  accessibilityLabel?: string;
+  accessibilityHint?: string;
+}
+
+export interface NativeStackHeaderBarButtonItemWithAction
+  extends SharedHeaderBarButtonItem {
+  /**
+   * Function to call when the button is pressed.
+   */
+  onPress: () => void;
+  /**
+   * A Boolean value that indicates whether the button is in a selected state.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/isselected
+   */
+  selected?: boolean;
+  /**
+   * A Boolean value that indicates whether the button represents an action or selection.
+   * Only available from iOS 15.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/changesselectionasprimaryaction
+   */
+  changesSelectionAsPrimaryAction?: boolean;
+}
+
+export interface NativeStackHeaderBarButtonItemMenuAction {
+  title?: string;
+  type: 'action';
+  onPress: () => void;
+  /**
+   * Any SF symbol. Explore them here: https://developer.apple.com/sf-symbols/
+   */
+  sfSymbolName?: string;
+  /**
+   * State of the menu item.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uimenuelement/state
+   */
+  state?: 'on' | 'off' | 'mixed';
+  /**
+   * Style of the menu item.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uimenuelement/attributes
+   */
+  attributes?: 'destructive' | 'disabled' | 'hidden' | 'keepsMenuPresented';
+  /**
+   * Discoverability title of the menu item.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uiaction/discoverabilitytitle
+   */
+  discoverabilityTitle?: string;
+}
+
+export interface NativeStackHeaderBarButtonItemSubmenu {
+  title?: string;
+  type: 'submenu';
+  /**
+   * Any SF symbol. Explore them here: https://developer.apple.com/sf-symbols/
+   */
+  sfSymbolName?: string;
+  items: NativeStackHeaderBarButtonItemWithMenu['menu']['items'];
+}
+export interface NativeStackHeaderBarButtonItemWithMenu
+  extends SharedHeaderBarButtonItem {
+  menu: {
+    title?: string;
+    items: (
+      | NativeStackHeaderBarButtonItemMenuAction
+      | NativeStackHeaderBarButtonItemSubmenu
+    )[];
+  };
+}
+
+export interface NativeStackHeaderBarButtonItemSpacing {
+  spacing: number;
+}
+
+export type NativeStackHeaderBarButtonItemWithCustomView<T> = {
+  customView: (props: T) => React.ReactNode;
+  /**
+   * A boolean value indicating whether the background this item may share with other items in the bar should be hidden.
+   * Only available from iOS 26.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/hidessharedbackground
+   */
+  hidesSharedBackground?: boolean;
+};
+
+export type NativeStackHeaderBarButtonItem =
+  | NativeStackHeaderBarButtonItemWithAction
+  | NativeStackHeaderBarButtonItemWithMenu
+  | NativeStackHeaderBarButtonItemSpacing;
```

Comment:
For consistency, remove `Bar` terminology - we just call it `Header` everywhere, e.g. `NativeStackHeaderButtonItem`

Reply:
https://github.com/react-navigation/react-navigation/pull/12657/commits/b6a0515a73f8cc7874ce0112549850111f8e81b5